tests/e2e/lib/capture_tcp.py

```
@@ -0,0 +1,53 @@
+    parser.add_argument("--output-dir", required=True)
+    return parser.parse_args()
+
+
+class CaptureHandler(socketserver.BaseRequestHandler):
+    def handle(self) -> None:
+        while True:
+            chunk = self.request.recv(65536)
+            if not chunk:
+                return
+            with self.server.lock:
+                with self.server.output_path.open("ab") as handle:
```

Ruff (0.15.10)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)